### PR TITLE
Reworks how body cameras function

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Icons for the worn overlay are taken from Partheo on Yogstation, <https://github.com/yogstation13/Yogstation/blob/master/icons/mob/clothing/neck/neck.dmi>, I made the bottom of the icons darker to stand out a little bit more.

Body cameras can now be put into any suit, but instead of following the user it follows the suit. It no longer shuts off when you take the suit off, but it also no longer automatically turns itself on, unless you're wearing the suit during the installation.

Since it can now be put into any suit, I gave it an overlay so it is visibly on the item/mob wearing one, rather than just having to assume every security suit has it.

Removes the body camera holder component, removes a ton of shitty and shortcut code, this is a much cleaner implementation.

https://github.com/user-attachments/assets/1c969f6e-d6d4-47fe-ace9-25db1cd1ba50

## Why It's Good For The Game

Suits can be taken off and not automatically shut itself off, turning the camera off is as easy as swiping your own ID on it, taking the camera out, or destroying the item.

### CODE REVIEW

File: fulp_modules/features/clothing/body_cameras/body_camera.dm [hunk 1]
```diff
@@ -1,54 +1,87 @@
...
+	 */
+	var/static/mutable_appearance/equipped_overlay = mutable_appearance('fulp_modules/features/clothing/body_cameras/bodycamera_overlay.dmi', "bodycamera")
 
 /obj/item/bodycam_upgrade/examine_more(mob/user)
 	. = ..()
-	. += list(span_notice("Use [src] on any valid vest to instantaneously install"))
-	. += list(span_notice("Use a [span_bold("screwdriver")] to remove it."))
-	. += list(span_notice("Once installed, the camera will activate/deactivate whenever its associated vest is worn/removed."))
-	. += list(span_notice("You can also use your ID card on the vest to toggle its camera manually."))
+	. += list(span_notice("You can use [name] on any vest to install it, or a [EXAMINE_HINT("screwdriver")] on the vest to uninstall one."))
```
Comment (QuiteLiterallyAnything) [inline fulp_modules/features/clothing/body_cameras/body_camera.dm:22]:
```suggestion
	. += list(span_notice("You can use [name] on any suit to install it."))
```
"Suit" might be accurate than "vest". Also the component itself already adds information on how to remove the camera.

File: fulp_modules/features/clothing/body_cameras/body_camera.dm [hunk 2]
```diff
@@ -57,114 +90,60 @@
...
-	var/mob/living/equipper = isliving(parent_item.loc) ? parent_item.loc : null
-	if(equipper && equipper.get_item_by_slot(clothingtype_required) == parent)
-		turn_camera_on(equipper)
-
-/// When the camera holder is unequipped
-/datum/component/bodycamera_holder/proc/on_unequip(mob/living/source, force, atom/newloc, no_move, invdrop, silent)
+/obj/item/bodycam_upgrade/proc/on_examine_more(atom/source, mob/user, list/examine_list)
 	SIGNAL_HANDLER
-	turn_camera_off()
+	examine_list += span_notice("It has [name] installed. You can remove it with a [EXAMINE_HINT("screwdriver")].")
```
Comment (QuiteLiterallyAnything) [inline fulp_modules/features/clothing/body_cameras/body_camera.dm:108]:
```suggestion
	examine_list += span_notice("It has [name] installed. You can toggle it with an [EXAMINE_HINT("ID card")] or remove it with a [EXAMINE_HINT("screwdriver")].")
```

File: fulp_modules/features/clothing/body_cameras/body_camera.dm (hunk 1 above)
Comment (JohnFulpWillard) [reply fulp_modules/features/clothing/body_cameras/body_camera.dm:22]: Suit can also mean jumpsuit

File: fulp_modules/features/clothing/body_cameras/body_camera.dm [hunk 3]
```diff
@@ -1,54 +1,87 @@
...
- * The bodycamera
+ * Bodycamera Upgrade
  *
- * This is the item that gets installed into items that have the bodycamera_holder element
+ * An item that can be inserted into any piece of clothing that goes on the Exosuit slot.
+ * Allows you to turn it on/off with an ID card to add the suit & the wearer's name into the security camera system.
  */
 /obj/item/bodycam_upgrade
 	name = "\improper body camera"
+	desc = "An armor vest upgrade, there's an instructions tag if you look a little closer..."
```
[the remark this answers is not in the document]
Comment (JohnFulpWillard) [reply fulp_modules/features/clothing/body_cameras/body_camera.dm:9]: Same about vests not suits. I think they are technically exosuits but I don't think that would be very descriptive to average players.

File: fulp_modules/features/clothing/body_cameras/body_camera.dm (hunk 1 above)
Comment (QuiteLiterallyAnything) [reply fulp_modules/features/clothing/body_cameras/body_camera.dm:22]: True. "Outer layer clothing" it is then.

File: fulp_modules/features/clothing/body_cameras/body_camera.dm (hunk 3 above)
[the remark this answers is not in the document]
Comment (QuiteLiterallyAnything) [reply fulp_modules/features/clothing/body_cameras/body_camera.dm:9]: Suggestion edited to use "outer layer clothing" instead.

**Pull request merged**